### DISCUSSION TITLE
Issue with Ipython and Kedro even when installation goes smoothely

### DISCUSSION THREAD

**Discussion by rosanajurdi** [community/community#63126] [Programming Help]
### Kedro Start Up and Installation 

Question

### Body

Hello,
I am running into an issue with kedro installation.Kedro installation seems to work fine with proper version and kedro info workings well but when I try to run kedro ipython I recieve the following error message:
Traceback (most recent call last):
  File "/home/usename/.local/lib/python3.10/site-packages/IPython/core/shellapp.py", line 282, in init_extensions
    self.shell.extension_manager.load_extension(ext)
  File "/home/usename/.local/lib/python3.10/site-packages/IPython/core/extensions.py", line 76, in load_extension
    return self._load_extension(module_str)
  File "/home/usename/.local/lib/python3.10/site-packages/IPython/core/extensions.py", line 91, in _load_extension
    mod = import_module(module_str)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'kedro'

Is anyone familiar with this error ? If so, how shoulkd I proceed.

This error arose when doing the spaceflights tutorial 

Thank you

**Comment by nirav-gajera** [comment]
To fix this error, you can try the following:

1. Make sure that you have installed the `kedro` module. You can do this by running the following command:

```
pip install kedro
```

2. Make sure that you are using the same environment that you used to install the `kedro` module. You can check the current environment by running the following command:

```
conda env list
```

If you are not using the same environment, you can activate the environment by running the following command:

```
conda activate <environment_name>
```

3. Make sure that there is no typo in the name of the `kedro` module. You can check the name of the module by running the following command:

```
pip show kedro
```

If the name of the module is correct, then you may need to restart your kernel. You can do this by pressing `Ctrl`+`C` in your IPython console and then typing `restart`.

**Reply by jakoblinski** [reply]
that do be crazy

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @rosanajurdi ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by SAMIULLAH-KALHORO-DEV** [comment]
It looks like you're encountering an issue with the Kedro installation and its integration with IPython. The error message indicates that IPython is unable to find the 'kedro' module, which suggests that Kedro might not be properly installed or configured in your environment. Let's go through some troubleshooting steps to address this issue:

1. **Check Kedro Installation:** First, ensure that Kedro is correctly installed. You mentioned that the installation seemed to work fine, but it's worth confirming. Open your terminal and run the following command to see if Kedro is installed:

   ```
   kedro info
   ```

   If you receive information about Kedro's version and other details, then Kedro is installed correctly.

2. **Activate Virtual Environment:** If you're using a virtual environment to manage your Python packages, make sure it's activated. Sometimes, issues arise when packages are installed in one environment but the wrong one is being used.

3. **IPython Compatibility:** Ensure that IPython is compatible with Kedro. Kedro might be expecting a specific version of IPython. You can check the recommended IPython version for your installed Kedro version in the Kedro documentation.

4. **Kernel Restart:** If you're using Jupyter Notebook or JupyterLab, try restarting the kernel for the notebook you're working on. Sometimes, the kernel might not recognize newly installed packages until it's restarted.

5. **Kernel Environment:** Make sure that the kernel used by IPython or Jupyter is associated with the same environment where Kedro is installed. You can check this by running the following command in a Jupyter Notebook cell:

   ```python
   import sys
   print(sys.executable)
   ```

   This will show you the Python executable path associated with the kernel. Compare it with the Python executable path where you installed Kedro.

6. **Reinstall Kedro:** If none of the above steps work, consider reinstalling Kedro. You can do this by first uninstalling Kedro and then reinstalling it:

   ```
   pip uninstall kedro
   pip install kedro
   ```

7. **Kedro Tutorial:** If the issue persists specifically in the context of the "spaceflights" tutorial, there might be an issue with the tutorial itself or its setup. You could try reaching out to the tutorial maintainers or the Kedro community for assistance.

8. **Python Version:** Ensure that you're using a compatible Python version for Kedro. Check the Kedro documentation for the recommended Python version.

If you've gone through these troubleshooting steps and the issue still persists, it might be helpful to provide more context about your environment (Python version, virtual environment setup, etc.) so that someone can offer more targeted assistance.
